app/workers/parsing.rb

```diff
@@ -108,7 +107,9 @@ def insert_into_user_snps
           #{temp_table_name}.local_genotype,
           #{genotype.id} as genotype_id
         from #{temp_table_name}
-        where #{temp_table_name}.snp_name not in (#{known_user_snps.to_sql})
+        where #{temp_table_name}.snp_name not in (
+          select snp_name from user_snps where genotype_id = #{genotype.id}
+        )
```

Comment:
I made this more explicit, so it's more clear that all the work is done by the database.